_posts/2017-12-05-book-review-docker-up-running.md

```diff
@@ -0,0 +1,117 @@
+Swarm, Centurion and Amazon EC2 container service. I am confused why the book is
+missing Kubernets and Helios here. As a part of advanced topics, you will find a
+comparison of various filesystems with a shallow overview of how Docker engine
+interacts with them. The same chapter is discussing available execution drivers
+and introduces LXC as another container technology. This API option is
+deprecated by Docker version 17.0.9 which makes libcontainer as the only
+dependency. I got to aware of how Docker containers are providing the
+virtualization layer using Namespaces. Docker limits the execution of container
+using CGroups (Control Groups). Namespaces and CGroups are GNU/Linux level
+dependencies used by Docker under the hood. If you are an API developer, then
+you should not skip Chapter 11. This chapter is discussing two well-followed
+patterns Twelve-Factor App and The Reactive manifesto. These guidelines are
+helpful while designing the architecture of your services. The book concludes
+with further challenges of using Docker as a container tool.
+
+One typo I found at page number 123, second last line, 'expore some of the
+tools...', here 'expore' is a typo and it should be 'explore some of the
+tools...'. I have submitted it to the official errata. At the time of writing
+this post it has not confirmed by the authority. It seems errata requests are
+not frequently reviewed. Hope responsible will confirm it soon.
+
+
+Who should read this book?
+
+* If you are developer who is unaware of the Docker or who is using Docker but
+  not aware of the architecture of it then I must advise to read this book.
+
+* If your job is to setup deployment cluster using Docker then this book can
+  help you to get informed about the Linux host internals. It will be helpful to
+  get aware of various debugging strategies and security guidelines for
+  container infrastructure. Good to get aware of how Docker engine is
+  communicating with your host workstation.
+
+* This book is not a reference book. If you already well aware of Docker, then I
+  will not advise you to not read this book. In such case, Docker documentation
+  can be the best reference.
+
+* I am assuming this book written, Docker was not supporting Windows platform
+  natively. The book purely focuses on GNU/Linux platform and highlight ways.
+  It highlights ways to run Docker over Windows using VM and Boot2Docker for
+  Non-Linux VM-based servers.
+
+
+What to keep in mind?
+
+* Docker is changing rapidly. There will be situations where mentioned options
+  are deprecated. In such situation, you have to browse the latest Docker
+  documentation and try to follow them.
+
+* You will be able to understand the official documentation better after reading
+  this book.
+
+
+Conclusion
+
+* Your GNU/Linux skills are your Docker skills. Once you will understand what
+  internally Docker is using, your decisions will become more mature.
```

Comment:
Once you understand what Docker is using internally, your decisions will become better.